src/v/cloud_storage/partition_manifest.cc

```diff
@@ -230,11 +232,49 @@ model::initial_revision_id partition_manifest::get_revision_id() const {
 
 remote_segment_path partition_manifest::generate_segment_path(
   const partition_manifest::key& key, const segment_meta& meta) const {
-    auto name = generate_segment_name(key.base_offset, key.term);
-    return generate_remote_segment_path(
+    auto name = cloud_storage::generate_local_segment_name(
+      key.base_offset, key.term);
+    return cloud_storage::generate_remote_segment_path(
       _ntp, meta.ntp_revision, name, meta.archiver_term);
```

Comment:
This doesn't look right - we use a local name in the remote path. Should we just use `partition_manifest::generate_remote_segment_path` everywhere?